docs/example_navigation_ros_codegen.md

```diff
@@ -0,0 +1,527 @@
+        mpc_controllerSolverStatus status = solve(); /* solve!  */
+        updateResults(status); /* pack results into `results` */
+        publishToTopic(publisher);
+    }
+
+    void CommandPoseCallback(const geometry_msgs::PoseStampedConstPtr& msg)
+    {
+        current_ref[0] = msg->pose.position.x;
+        current_ref[1] = msg->pose.position.y;
+        current_ref[2] = msg->pose.orientation.w;  // we will use this field for the yaw in degrees.
+    }
+
+    void odometryCallback(const nav_msgs::Odometry::ConstPtr &msg) {
+        double roll, pitch, yaw;
+
+        tf::Quaternion quat(msg->pose.pose.orientation.x, msg->pose.pose.orientation.y, msg->pose.pose.orientation.z, msg->pose.pose.orientation.w);
+        tf::Matrix3x3(quat).getRPY(roll, pitch, yaw);
+
+        current_pos[0] = msg->pose.pose.position.x;
+        current_pos[1] = msg->pose.pose.position.y;
+        current_pos[2] = yaw;
+    }
+
+    void solveAndPublishCmdVel(ros::Publisher& publisher)
+    {
+        double current_par [MPC_CONTROLLER_NUM_PARAMETERS] = {0};
+        double current_var [MPC_CONTROLLER_NUM_DECISION_VARIABLES] = {0};
+        double lin_vel_cmd, ang_vel_cmd = 0;
+
+        current_par[0] = current_pos[0];
+        current_par[1] = current_pos[1];
+        current_par[2] = current_pos[2];
+        current_par[3] = current_ref[0];
+        current_par[4] = current_ref[1];
+        current_par[5] = current_ref[2];
+
+        /* solve                  */
+        mpc_controllerSolverStatus status = mpc_controller_solve(cache, current_var, current_par, 0, &init_penalty);
+
+        lin_vel_cmd = current_var[0];
+        ang_vel_cmd = current_var[1];
+
+        geometry_msgs::Twist twist;
+        twist.linear.x = lin_vel_cmd;
+        twist.linear.y = 0.0;
+        twist.linear.z = 0.0;
+        twist.angular.x = 0.0;
+        twist.angular.y = 0.0;
+        twist.angular.z = ang_vel_cmd;
+        publisher.publish(twist);
+
+        ROS_INFO("x: %f, y: %f, yaw: %f", current_pos[0], current_pos[1], current_pos[2]);
+        ROS_INFO("Solve time: %f ms. I will send %f %f \n", (double)status.solve_time_ns / 1000000.0, lin_vel_cmd, ang_vel_cmd);
+
+    }
+
+}; /* end of class OptimizationEngineManager */
+
+} /* end of namespace open_nmpc_controller */
+
+/**
+ * Main method
+ *
+ * This advertises a new (private) topic to which the optimizer
+ * announces its solution and solution status and details. The
+ * publisher topic is 'open_nmpc_controller/result'.
+ *
+ * It obtains inputs from 'open_nmpc_controller/parameters'.
+ *
+ */
+int main(int argc, char** argv)
+{
+
+    std::string result_topic, params_topic;  /* parameter and result topics */
+    std::string out_twist, in_odometry;
+    double rate; /* rate of node (specified by parameter) */
+
+    open_nmpc_controller::OptimizationEngineManager mng;
+    ros::init(argc, argv, ROS_NODE_MPC_CONTROLLER_NODE_NAME);
+    ros::NodeHandle private_nh("~");
```

Comment:
Don't you need both a private and a global node here? (where the global node is used to subscribe to the topics of the various sensors).